zebra/zebra_vty.c

```diff
@@ -2041,6 +2044,213 @@ DEFPY (show_route_summary,
 	return CMD_SUCCESS;
 }
 
+DEFUN_HIDDEN (show_route_zebra_dump,
+              show_route_zebra_dump_cmd,
+              "show <ip|ipv6> zebra route dump [<vrf> VRFNAME]",
+              SHOW_STR
+              IP_STR
+              IP6_STR
+              "Zebra daemon\n"
+              "Routing table\n"
+              "All information\n"
+              VRF_CMD_HELP_STR)
+{
+	afi_t afi;
+	struct route_table *table;
+	const char *vrf_name = NULL;
+	int idx = 0;
+
+	if (argv_find(argv, argc, "ip", &idx))
+		afi = AFI_IP;
+
+	if (argv_find(argv, argc, "ipv4", &idx)
+	    || argv_find(argv, argc, "ipv6", &idx))
+		afi = strmatch(argv[idx]->text, "ipv6") ? AFI_IP6 : AFI_IP;
+
+	if (argv_find(argv, argc, "vrf", &idx))
+		vrf_name = argv[++idx]->arg;
+
+	if (!vrf_name) {
+		struct vrf *vrf;
+		struct zebra_vrf *zvrf;
+
+		RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
+			zvrf = vrf->info;
+			if ((zvrf == NULL)
+			    || (zvrf->table[afi][SAFI_UNICAST] == NULL))
+				continue;
+
+			table = zvrf->table[afi][SAFI_UNICAST];
+			show_ip_route_dump_vty(vty, table);
+		}
+	} else {
+		vrf_id_t vrf_id = VRF_DEFAULT;
+
+		if (vrf_name)
+			VRF_GET_ID(vrf_id, vrf_name, true);
+
+		table = zebra_vrf_table(afi, SAFI_UNICAST, vrf_id);
+		if (!table)
+			return CMD_SUCCESS;
+
+		show_ip_route_dump_vty(vty, table);
+	}
+
+	return CMD_SUCCESS;
+}
+
+static void show_ip_route_nht_dump(struct vty *vty, struct nexthop *nexthop,
+				   struct route_entry *re, unsigned int num)
+{
+
+	char buf[SRCDEST2STR_BUFFER];
+
+	vty_out(vty, "   Nexthop %u :\n", num);
+	vty_out(vty, "      type : %u\n", nexthop->type);
+	vty_out(vty, "      flags : %u\n", nexthop->flags);
+
+	switch (nexthop->type) {
+	case NEXTHOP_TYPE_IPV4:
+	case NEXTHOP_TYPE_IPV4_IFINDEX:
+		vty_out(vty, "      ip address: %s\n",
+			inet_ntoa(nexthop->gate.ipv4));
+		vty_out(vty, "      afi : ipv4\n");
+
+		if (nexthop->ifindex) {
+			vty_out(vty, "      interfaceIndex : %d\n",
+				nexthop->ifindex);
+			vty_out(vty, "      interfaceName : %s\n",
```

Comment:
As before with `:`. Also, why are these camelCased? This isn't machine friendly output, so I assume it's for humans - in which case it's clearer to write "Interface index" and "Interface name"

Reply:
Done